### DISCUSSION TITLE
Datasmith View Issue

### DISCUSSION THREAD

**Discussion by hinchmonster** [iTwin/itwinjs-core#4597] [Q&A]
When I attempt to export the itwin model with a specific view it fails to export any meshes and textures. Only when I export the whole model it works. Is this a bug or am I missing something?

**Comment by arnab-git-hub** [comment]
@MichaelBelousov Can you please look into this issue?

**Comment by MichaelBelousov** [comment]
@hinchmonster this sounds like a bug I recently encountered and I am working on deploying a fix, hopefully this week.

**Reply by MichaelBelousov** [reply]
@hinchmonster 0.7.3 has been released with a potential fix... can you try updating?

**Reply by hinchmonster** [reply]
@MichaelBelousov thanks very much this fix has fixed the animation issues but the view issues are still there.

When you export with a view filter, it exports now which it didnt in the last version but it ignores the view filter completely

**Reply by MichaelBelousov** [reply]
@hinchmonster is it possible for you to share with me your .sp file (privately send to mike.belousov@bentley.com) so that I can reproduce and work on a fix?

**Reply by hinchmonster** [reply]
@MichaelBelousov sent you an email there thanks

**Reply by MichaelBelousov** [reply]
@hinchmonster I have identified a problem in this case when exporting using the "export current view" button. We are working on deploying a fix in the next few days, is that the techniqueyou used to export? Or was it something else? Can you give me precise instructions on how to filter the iModel for export, or use the `download viewport` option in the saved views widget and email me it to me?

**Reply by hinchmonster** [reply]
there is an issue with current view where it does export anything but there is also an issue with a saved view where by it exports the model but the animation fails and the meshes hidden in the view still export just without textures.

**Reply by MichaelBelousov** [reply]
I will see if I can reproduce that on your data

**Comment by hinchmonster** [comment]
@MichaelBelousov also something not sure if it is an unreal issue of the datasmith issues the shadows from the animation are always there even before the object has appeared is this a me issue or an datasmith issue?

**Reply by MichaelBelousov** [reply]
see this section of our webinar on the iTwin Exporter for Datasmith: https://youtu.be/IV0eBECws1k?t=2510
You want to disable "cast ray traced shadows" on your transparent materials.
If you could ask this as a separate question we can mark this as the answer there so it doesn't interfere with this question.

**Reply by MichaelBelousov** [reply]
@hinchmonster can you confirm that this was an issue with your data?
